tensorflow_addons/image/distance_transform.py

```diff
@@ -0,0 +1,84 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Distance transform ops."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.python.framework import common_shapes
+from tensorflow.python.framework import ops
+from tensorflow_addons.utils.resource_loader import get_path_to_datafile
+
+_image_ops_so = tf.load_op_library(
+    get_path_to_datafile("custom_ops/image/_image_ops.so"))
+
+ops.NotDifferentiable("EuclideanDistanceTransform")
+ops.RegisterShape("EuclideanDistanceTransform")(
+    common_shapes.call_cpp_shape_fn)
+
+
+@tf.function
+def euclidean_dist_transform(images,
+                             dtype=tf.dtypes.float32,
+                             name="euclidean_distance_transform"):
+    """
+    Applies euclidean distance transform to the images_t
+
+    Args:
+      images: Tensor of shape (num_images, num_rows, num_columns, num_channels)
+        (NHWC) or (num_rows, num_columns, num_channels) (HWC). The rank must be
+        statically known. The image(s) must be a binary or grayscale
+        of uint8 type.
+      dtype: The dtype of the output, must be float16, float32 or float64.
+      name: The name of the op.
+
+
+    Returns:
+      Image(s) with the same type and shape as `images`, with euclidean
+      distance transform applied.
+
+    Raises:
+      TypeError: If `image` is not uint8, or `dtype` is not floating point.
+      ValueError: If `image` is not a binary or grayscale image of rank 3 or 4.
+
+    """
+    with tf.name_scope(name):
+        image_or_images = ops.convert_to_tensor(images, name="images")
+
+        if image_or_images.dtype.base_dtype != tf.dtypes.uint8:
+            raise TypeError(
+                "Invalid dtype %s. Excepted uint8." % image_or_images.dtype)
+        elif image_or_images.get_shape().ndims is None:
+            raise ValueError("`images` rank must be statically known")
+        elif len(image_or_images.get_shape()) == 3:
+            images = image_or_images[None, :, :, :]
+        elif len(image_or_images.get_shape()) == 4:
+            images = image_or_images
+        else:
+            raise ValueError("`images` should have rank between 3 and 4")
+
+        if images.get_shape()[3] != 1 and images.get_shape()[3] is not None:
+            raise ValueError("`images` must have only one channel")
+
+        if not dtype.is_floating:
```

Comment:
By the way, can we use `if dtype not in [tf.float16, tf.float32, tf.float64]`? Because we only create test cases for them.